### PR TITLE
Context manager for temprary config changes

### PR DESCRIPTION
**Pull request by eslickj**
This adds a context manager that allows you to temporarily change the IDAES config and have it revert back after the with-block.  This is mainly for testing.  Config changes for one test can affect others.  For example, you may want to set ipopt to default to user scaling, so all the initialization routines use it.  But that setting will stick around for subsequent tests potentially breaking them.  So the idea is that you can use the new context manager to make sure tests don't make lasting changes to the config.

This also makes some changes so the pointer to the IDAES ConfigDict is only in one place. 

### Legal Acknowledgement

By contributing to this software project, I agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the license terms described in the LICENSE.txt file at the top level of this directory.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

### CODE REVIEW

**Review by andrewlee94 — APPROVED** [review]
Looks OK to me, although I don't fully understand some of it.

File: idaes/tests/test_ipopt.py [hunk 1]
```diff
@@ -36,19 +36,24 @@ def test_ipopt_idaes_config():
...
+    #
+    # TODO(JCE): Bring back the rest of this test after the tests
+    # untangled in the GT PR
+    #
     # back to the original config, don't use idaes default option settings
-    solver = pyo.SolverFactory('ipopt')
-    assert "nlp_scaling_method" not in solver.options
-    assert solver.options["tol"] != 1
-
+    #solver = pyo.SolverFactory('ipopt')
```
Comment (andrewlee94) [inline idaes/tests/test_ipopt.py:52]: Should this comment code still be here?

**Comment by eslickj** [comment]
@andrewlee94, could you approve again?  I didn't really change anything since you approved, I just had to comment out part of a test, until we get some of the fragile tests fixed.  As it is now some test depend on other tests re-configuring the solvers.  I sorted that out in PR #323, so I'll bring back the rest of the test there.

**Review by MAZamarripa — APPROVED** [review]
lgtm

**Pull request merged**